<a href="https://colab.research.google.com/github/juliafoliveira-tech/Bioestat-stica/blob/main/tarefa_12.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

* Julia Feitosa NºUSP: 16228410
* Raissa Pereira Lima da Silva NºUSP: 14605077
* Stéfani Machado NºUSP: 15291452


ATIVIDADE SEMANA 8 Probabilidade em Biologia e Ecologia, distribuições Bernoulli, Binomial e Poisson

Objetivo: reconhecer e aplicar as distribuições de Bernoulli, Binomial e Poisson em situações biológicas e ecológicas.

QUESTÃO 1. Qual modelo usar? Para cada situação, indique qual distribuição é mais adequada: Bernoulli, Binomial ou Poisson. Justifique brevemente.

A. Um pesquisador verifica se uma determinada semente germinou ou não.

R: Bernoulli. Trata-se de um único ensaio (uma única semente testada) que apresenta apenas dois resultados possíveis mutuamente exclusivos: sucesso (germinou) ou fracasso (não germinou).

B. De 20 sementes plantadas, deseja-se calcular a probabilidade de exatamente 15 germinarem, considerando que a probabilidade de germinação de cada semente é 0,8.

R: Binomial. Envolve uma quantidade fixa de múltiplos ensaios independentes (20 sementes), cada um com a mesma probabilidade constante de sucesso (0,8), com o objetivo de calcular a probabilidade de obter um número exato de sucessos (15 germinações).

C. Em uma área de mata, são observados, em média, 4 ninhos de uma determinada espécie por hectare. Deseja-se calcular a probabilidade de encontrar exatamente 6 ninhos em um hectare.

R: Poisson. É utilizada para modelar a contagem de eventos raros ou discretos em um intervalo contínuo de espaço (por hectare), dado uma taxa média conhecida (média de 4 ninhos por hectare).

Para a situação de um pesquisador capturando aleatoriamente 10 abelhas, onde cada abelha pode ou não ter a característica genética (com 30% de probabilidade de ter), a **Distribuição Binomial** é a mais adequada.

**Razão:**

*   **Número fixo de tentativas (n):** O pesquisador captura 10 abelhas, então n = 10.
*   **Dois resultados possíveis (sucesso/fracasso):** Cada abelha ou 'apresenta a característica' (sucesso) ou 'não apresenta a característica' (fracasso).
*   **Probabilidade constante de sucesso (p):** A probabilidade de uma abelha apresentar a característica é de 30% (p = 0.30) e é constante para cada abelha capturada.
*   **Tentativas independentes:** A característica de uma abelha não afeta a característica de outra abelha capturada.

A distribuição binomial permite calcular a probabilidade de obter um certo número de 'sucessos' (abelhas com a característica genética) em um número fixo de tentativas.

### Questão 2
Na população de insetos (10 min)
Em uma população de abelhas, 30% dos indivíduos apresentam uma determinada
característica genética.
Um pesquisador captura aleatoriamente 10 abelhas.

## Diagrama de Árvore: Característica Genética em Abelhas

Para organizar essa situação em um diagrama de árvore, podemos começar com um nó raiz representando a população total de abelhas e, em seguida, ramificar com base na característica genética:

1.  **Nó Raiz:** População de Abelhas

2.  **Primeiro Nível de Ramificação:**
    *   **Ramo 1:** "Característica Genética Presente"
        *   **Probabilidade:** 30% (ou 0.30)
    *   **Ramo 2:** "Característica Genética Ausente"
        *   **Probabilidade:** 70% (ou 0.70)

**Visualização:**

```mermaid
graph TD
    A[População de Abelhas] -->|30%| B[Característica Presente]
    A -->|70%| C[Característica Ausente]
```

Este diagrama de árvore simples mostra a proporção de abelhas que apresentam ou não a característica genética mencionada.

Questões
a) Qual distribuição deve ser utilizada? Explique por quê.
A melhor distribuição é a binominal, ela é a mais adequada pois a análise possui número fixo de tentativas, os resultados são binários (ou a abelha tem ou não tem a característica), além disso as tentativas são independentes.

b) Qual é a probabilidade de exatamente 3 abelhas apresentarem a característica?

In [ ]:
from scipy.stats import binom

n = 10  # Número total de abelhas capturadas (tentativas)
k = 3   # Número exato de abelhas com a característica (sucessos desejados)
p = 0.30 # Probabilidade de uma abelha apresentar a característica (30%)

# Calcula a probabilidade de P(X=3)
prob_exactamente_3 = binom.pmf(k, n, p)

print(f"A probabilidade de exatamente {k} abelhas apresentarem a característica é: {prob_exactamente_3:.4f}")

A probabilidade de exatamente 3 abelhas apresentarem a característica é: 0.2668


### Explicação Detalhada do Cálculo (P(X=3))

Para calcular a probabilidade de exatamente 3 abelhas apresentarem a característica genética, utilizamos a Função de Massa de Probabilidade (PMF) da distribuição binomial, que é implementada como `binom.pmf()` na biblioteca `scipy.stats`.

*   `n = 10`: Representa o número total de tentativas (o número de abelhas capturadas).
*   `k = 3`: Representa o número de sucessos desejados (o número de abelhas que queremos que apresentem a característica genética).
*   `p = 0.30`: Representa a probabilidade de sucesso em uma única tentativa (a probabilidade de uma abelha ter a característica genética).

A função `binom.pmf(k, n, p)` calcula a probabilidade de obter exatamente `k` sucessos em `n` tentativas, onde a probabilidade de sucesso em cada tentativa é `p`. A fórmula geral para a PMF binomial é:

$$P(X=k) = \binom{n}{k} \cdot p^k \cdot (1-p)^{n-k}$$

Onde:
*   $\binom{n}{k}$ é o coeficiente binomial, que representa o número de maneiras de escolher `k` sucessos de `n` tentativas, dado por $\frac{n!}{k!(n-k)!}$.
*   $p^k$ é a probabilidade de obter `k` sucessos.
*   $(1-p)^{n-k}$ é a probabilidade de obter `n-k` fracassos.

Neste caso, o cálculo foi:

$$P(X=3) = \binom{10}{3} \cdot (0.30)^3 \cdot (1-0.30)^{10-3}$$
$$P(X=3) = \frac{10!}{3!(10-3)!} \cdot (0.30)^3 \cdot (0.70)^7$$
$$P(X=3) = 120 \cdot 0.027 \cdot 0.0823543$$
$$P(X=3) \approx 0.2668$$

Isso significa que há aproximadamente 26.68% de chance de que exatamente 3 das 10 abelhas capturadas apresentem a característica genética.

c) Qual é a probabilidade de pelo menos 1 das 10 abelhas apresentar a característica?

In [ ]:
from scipy.stats import binom

n = 10  # Número total de abelhas capturadas
p = 0.30 # Probabilidade de uma abelha apresentar a característica

# Probabilidade de nenhuma abelha apresentar a característica (X=0)
prob_none = binom.pmf(0, n, p)

# Probabilidade de pelo menos 1 abelha apresentar a característica (1 - P(X=0))
prob_at_least_1 = 1 - prob_none

print(f"A probabilidade de pelo menos 1 das 10 abelhas apresentar a característica é: {prob_at_least_1:.4f}")

A probabilidade de pelo menos 1 das 10 abelhas apresentar a característica é: 0.9718


QUESTÃO 3. Na observação ecológica.

Durante levantamentos em uma área de cerrado, pesquisadores registram em média 2 encontros com uma determinada espécie de mamífero por hora. Questões.

a) Qual distribuição seria mais adequada para assumir nesse caso?

b) Qual é a probabilidade de ocorrerem exatamente 3 encontros em uma hora?

c) Qual é a probabilidade de nenhum encontro ocorrer durante uma hora?

d) Qual é a probabilidade de ocorrerem mais de 2 encontros em uma hora?

In [ ]:
from scipy.stats import poisson

# Definição do parâmetro lambda (taxa média de encontros por hora)
lam = 2

print("a) Distribuição adequada:")
print("A distribuição mais adequada é a Distribuição de Poisson, pois modela o número de eventos discretos (encontros) que ocorrem em um intervalo contínuo de tempo (1 hora) com uma taxa média conhecida (2 encontros/hora).\n")

# b) Probabilidade de exatamente 3 encontros em 1 hora
p_3 = poisson.pmf(3, lam)
print(f"b) A probabilidade de ocorrerem exatamente 3 encontros é: {p_3:.4f} (ou {p_3 * 100:.2f}%)")

# c) Probabilidade de nenhum encontro em 1 hora
p_0 = poisson.pmf(0, lam)
print(f"c) A probabilidade de nenhum encontro ocorrer é: {p_0:.4f} (ou {p_0 * 100:.2f}%)")

# d) Probabilidade de ocorrerem mais de 2 encontros em 1 hora: P(X > 2) = 1 - P(X <= 2)
p_gt_2 = 1 - poisson.cdf(2, lam)
print(f"d) A probabilidade de ocorrerem mais de 2 encontros é: {p_gt_2:.4f} (ou {p_gt_2 * 100:.2f}%)")

### **Descrição Detalhada dos Cálculos (Questão 3)**

A distribuição de Poisson é definida pela sua função de probabilidade:

$$P(X = k) = \frac{e^{-\lambda} \cdot \lambda^k}{k!}$$

Onde:
*   $\lambda = 2$ é a taxa média de encontros por hora.
*   $e \approx 2,71828$ é a base do logaritmo natural.
*   $k$ é o número de encontros de interesse.
*   $k!$ representa o fatorial de $k$.

---

#### **b) Exatamente 3 encontros em uma hora ($k = 3$)**

Aplicando os valores na fórmula:

$$P(X = 3) = \frac{e^{-2} \cdot 2^3}{3!} = \frac{e^{-2} \cdot 8}{6}$$

$$P(X = 3) \approx \frac{0,135335 \cdot 8}{6} \approx \frac{1,08268}{6} \approx 0,1804 \text{ (ou 18,04\%)}$$

---

#### **c) Nenhum encontro em uma hora ($k = 0$)**

Aplicando os valores na fórmula (lembrando que $2^0 = 1$ e $0! = 1$):

$$P(X = 0) = \frac{e^{-2} \cdot 2^0}{0!} = \frac{e^{-2} \cdot 1}{1} = e^{-2}$$

$$P(X = 0) \approx 0,1353 \text{ (ou 13,53\%)}$$

---

#### **d) Mais de 2 encontros em uma hora ($P(X > 2)$)**

Para calcular a probabilidade de ocorrerem mais de 2 encontros, usamos a probabilidade complementar (1 menos a probabilidade de ocorrerem 2 ou menos encontros):

$$P(X > 2) = 1 - P(X \le 2)$$

$$P(X \le 2) = P(X = 0) + P(X = 1) + P(X = 2)$$

Calculando cada termo:
*   $P(X = 0) \approx 0,135335$
*   $P(X = 1) = \frac{e^{-2} \cdot 2^1}{1!} \approx 0,135335 \cdot 2 \approx 0,270670$
*   $P(X = 2) = \frac{e^{-2} \cdot 2^2}{2!} = \frac{e^{-2} \cdot 4}{2} \approx 0,135335 \cdot 2 \approx 0,270670$

Somando as probabilidades:

$$P(X \le 2) \approx 0,135335 + 0,270670 + 0,270670 = 0,676675$$

Finalmente, calculamos o complemento:

$$P(X > 2) = 1 - 0,676675 = 0,3233 \text{ (ou 32,33\%)}$$


4. Desafio final — Bernoulli versus Binomial versus Poisson?

Considere o seguinte experimento:
Um ecólogo examina uma árvore e verifica se ela está infestada por uma determinada espécie de inseto.


a) Qual é a distribuição se ele examinar uma única árvore?

Se ele examinar uma única árvore, a distribuição adequada é a **Bernoulli**.

Um experimento de Bernoulli tem apenas dois resultados  (infestada ou não infestada), e a probabilidade de sucesso permanece a mesma para cada tentativa.

b) Qual é a distribuição se ele examinar 50 árvores e quiser saber quantas estão infestadas?

* Para examinar 50 árvores e entender quantas estão infestadas, a distribuição adequada é a **Binomial**, visto que ela descreve o número de sucessos em um número fixo de tentativas independentes de Bernoulli, onde cada tentativa tem a mesma probabilidade de sucesso.
Neste caso, n = 50 árvores (tentativas) e cada uma pode ou não estar infestada (sucesso/fracasso).

c) Qual é a principal diferença entre os dois modelos (Bernoulli e Binomial)?
*   A **Distribuição de Bernoulli** é referente a um **único** evento (ou tentativa) com dois resultados possíveis (sucesso/fracasso, por exemplo). A **Distribuição Binomial** modela o **número de sucessos** em uma **sequência de múltiplos eventos independentes de Bernoulli** (um número fixo de tentativas).

Para discutir com a turma.

Complete as frases

*   **Bernoulli** é adequada quando interpreta-se o resultado de um único evento comm apenas duas possibilidades (sucesso ou fracasso).

*   **Binomial** é adequada quando conta-se o número de sucessos em um número fixo de tentativas independentes, onde cada tentativa tem dois resultados possíveis e a mesma probabilidade de sucesso. Ela é uma extensão do método de Bernoulli.

*   **Poisson** é adequada quando analisa-se o número de ocorrências de um evento em um intervalo fixo de tempo ou espaço, e eles ocorrem com uma taxa média conhecida e independentemente do tempo desde o último evento.